spec.bs

```diff
@@ -964,31 +978,95 @@ partial interface PrivateAggregation {
 
 Issue: Do we want to align naming with implementation?
 
-The {{Navigator/runAdAuction()}} method steps are modified to add the following
-steps after step 6 ("Let <var ignore>p</var> be [=a new promise=]"), renumbering
-later steps as appropriate:
-<div algorithm="protected-audience-runadauction-monkey-patch">
-7. Let |batchingScopeMap| be a new [=map/is empty|empty=] [=map=].
+<div algorithm>
+The <dfn method for="PrivateAggregation">contributeToHistogramOnEvent(DOMString
+event, PAExtendedHistogramContribution contribution)</dfn> method steps are:
+</div>
+1. Let |scopingDetails| be the {{PrivateAggregation}} object's
+    [=PrivateAggregation/scoping details=].
+1. If |scopingDetails| is null, throw a new {{DOMException}} with name
+    "`NotAllowedError`".
+1. If |event| [=string/starts with=] "`reserved.`" and « "`reserved.always`",
+    "`reserved.loss`", "`reserved.win`" » does not [=list/contain=] |event|,
+    return.
+
+    Note: No error is thrown to allow forward compatibility if additional
+        reserved event types are added later.
+1. Let |bucket| be |contribution|["{{PAExtendedHistogramContribution/bucket}}"].
+1. If |bucket| is a {{PASignalValue}}:
+    1. If |bucket|["{{PASignalValue/baseValue}}"] is not a valid [=signal base
+        value=], throw a {{TypeError}}.
+    1. If |bucket|["{{PASignalValue/offset}}"] is a {{long}}, throw a
+        {{TypeError}}.
+1. Let |value| be |contribution|["{{PAExtendedHistogramContribution/value}}"].
+1. If |value| is a {{PASignalValue}}:
+    1. If |value|["{{PASignalValue/baseValue}}"] is not a valid [=signal base
+        value=], throw a {{TypeError}}.
+    1. If |value|["{{PASignalValue/offset}}"] is a {{bigint}}, throw a
+        {{TypeError}}.
+1. Let |batchingScope| be null.
+1. If |event| [=string/starts with=] "`reserved.`", set |batchingScope| to the
+    result of running |scopingDetails|' [=scoping details/get batching scope
+    steps=].
+
+    Note: Each non-reserved |event| will have a different [=batching scope=]
+        that is created later.
+1. Let |entry| be a new [=on event contribution cache entry=] with the items:
+    : [=on event contribution cache entry/contribution=]
+    :: |contribution|
+    : [=on event contribution cache entry/batching scope=]
+    :: |batchingScope|
+    : [=on event contribution cache entry/debug scope=]
+    :: The result of running |scopingDetails|' [=scoping details/get debug scope
+        steps=].
+    : [=on event contribution cache entry/worklet function=]
+    :: The [=worklet function=] that is currently being executed.
+
+1. Let |auctionConfig| be the <a spec="turtledove">auction config</a> associated
+    with the global scope.
+1. Let |ig| be null.
+1. If this function execution is associated with a bidder, set |ig| to the
+    [=interest group=] associated with the current function execution.
```

Comment:
High level wise, this is fine.
Spec wise, this can be written as something similar to [sendReportTo](
https://wicg.github.io/turtledove/#dom-interestgroupreportingscriptrunnerglobalscope-sendreportto), like `this's relevant global object's's [=InterestGroupBiddingScriptRunnerGlobalScope/interest group=]`

Reply:
See comment below, but yeah that makes sense. Will follow up